crates/net/network/src/session/active.rs

```diff
@@ -746,14 +787,47 @@ impl<N: NetworkPrimitives> Future for ActiveSession<N> {
                 };
 
                 if should_send {
-                    this.queued_outgoing.push_back(
-                        EthMessage::BlockRangeUpdate(this.local_range_info.to_message()).into(),
-                    );
+                    // Prefer eth/70 SendBlockRange; fall back to eth/69 BlockRangeUpdate.
+                    let msg = if this.conn.version() >= EthVersion::Eth70 {
+                        EthMessage::SendBlockRange(RequestPair {
+                            request_id: 0,
+                            message: SendBlockRange::from_range(BlockRange {
+                                start_block: this.local_range_info.earliest(),
+                                end_block: this.local_range_info.latest(),
+                            }),
+                        })
+                    } else {
+                        EthMessage::BlockRangeUpdate(this.local_range_info.to_message())
+                    };
+                    this.queued_outgoing.push_back(msg.into());
                     this.last_sent_latest_block = Some(current_latest);
                 }
             }
         }
 
+        if this.conn.version() >= EthVersion::Eth70 {
+            // If our view of the peer's range is stale, request an update (throttled).
+            let now = Instant::now();
```

Comment:
I think this is a bit too expensive,

can we combine this with this interval here:

https://github.com/paradigmxyz/reth/blob/190297083a3d82221c6ecb557d5a14026f719651/crates/net/network/src/session/active.rs#L737-L739